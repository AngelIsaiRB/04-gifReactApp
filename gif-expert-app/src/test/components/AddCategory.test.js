const { AddCategory } = require("../../components/AddCategory");
import { shallow } from 'enzyme';
import React, { PureComponent } from 'react'


describe('pruebas en AddCategory', () => {
    // const wrapper = 
    const setCategories=jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} /> );

    beforeEach(()=>{
        jest.clearAllMocks();
         wrapper = shallow(<AddCategory setCategories={setCategories} /> );
    });

    test('should de de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should simular cuando la caja de texto cambia', () => {
        const input = wrapper.find("input");
        const value= "Holamundo";
        input.simulate("change",{target:{value:value}});
        expect(wrapper.find("p").text().trim()).toBe(value);
    });

    test('should No debe de posterar la informacin con submit', () => {
        wrapper.find("form").simulate("submit",{
            preventDefault(){}
        });
        expect(setCategories).not.toHaveBeenCalled();
        
    })
    
    
});
