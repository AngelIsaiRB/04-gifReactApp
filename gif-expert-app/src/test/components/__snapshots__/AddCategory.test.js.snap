// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en AddCategory should de de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
